espresso-support/demo/build.gradle

```diff
@@ -36,11 +36,17 @@ android {
 }
 
 dependencies {
-    androidTestCompile project(':lib')
-    androidTestCompile(librariesTest.androidTestRunner) {
+    debugCompile project(':extras')
```

Comment:
This contains the ViewActivity so it needs to be compiled into the apk under test. This might be debug sourceset or you might have a specific sourceset/build flavor for espresso.

The `core` artifact is the one that's explicitly only needed for `androidTestCompile`